pilot/pkg/autoregistration/controller.go

```diff
@@ -520,6 +525,43 @@ func sanitizeIP(s string) string {
 	return strings.ReplaceAll(s, ":", "-")
 }
 
+// workloadentryinfo just represents the name/namespace of a workloadentry, the minimum information required
+// for a delete
+type workloadentryinfo struct {
+	name      string
+	namespace string
+}
+
+func (c *Controller) onWorkloadEntryEvent(_, curr config.Config, event model.Event) {
+	log.Infof("Handling event %s for workload entry %s/%s", event, curr.Namespace, curr.Name)
+	// if a workloadentry is created with the same IP address & Network as an already existing workloadentry, delete the old workloadentry.
+	entry := serviceentry.ConvertWorkloadEntry(curr)
+
+	if event == model.EventDelete {
+		// only delete if it is not a delete caused by stale WLE
+		deleteEntryValue := workloadentryinfo{curr.Name, curr.Namespace}
+		if pe, ok := c.workloadEntryStatus[makeWEkey(entry)]; ok && pe == deleteEntryValue {
+			delete(c.workloadEntryStatus, makeWEkey(entry))
+		}
+		return
+	} else if prevEntry, ok := c.workloadEntryStatus[makeWEkey(entry)]; ok && prevEntry.name != curr.Name {
+		// delete stale WLE
+		if err := c.store.Delete(gvk.WorkloadEntry, prevEntry.name, prevEntry.namespace, nil); err != nil && !errors.IsNotFound(err) {
```

Comment:
I do think this is not safe, think about this case, a user create a WLE 1 and then sometime later he creates another WLE 2, with same address.

Reply:
For auto registed wle, how could this happen?

Reply:
we only add it to the map if it has the AutoRegisted annotation. See 

```go
// only add to map if WorkloadEntry was auto-registered
	if _, ok := curr.Meta.Annotations[AutoRegistrationGroupAnnotation]; ok {
		c.workloadEntryStatus[makeWEkey(entry)] = config.NamespacedName(curr)
	}
```

Reply:
I was wondering how could auto register introduce overlapping

Reply:
my example is this:

I'm running 3 dedicated ztunnels, all with `INSTANCE_IP` set to `1.1.1.1`. I then auto-register them. Autoregistration uses the 1.1.1.1 IP and hence 3 WLEs are registed with the same IP.